client/logclient.go

```diff
@@ -35,6 +35,13 @@ type LogClient struct {
 	jsonclient.JSONClient
 }
 
+// CheckLogClient is an interface that allows (just) checking of various log contents.
+type CheckLogClient interface {
```

Comment:
As noted elsewhere if we didn't have this interface and the DNS code returned not implemented errors then it could make some things simpler.